app/assets/javascripts/discourse/app/lib/url.js

```diff
@@ -286,27 +286,23 @@ const DiscourseURL = EmberObject.extend({
 
   // Determines whether a URL is internal or not
   isInternal(url) {
-    if (url && url.length) {
-      if (url.startsWith("//")) {
-        url = "http:" + url;
-      }
-      if (url.startsWith("#")) {
-        return true;
-      }
-      if (url.startsWith("/")) {
-        return true;
-      }
-      if (url.startsWith(this.origin())) {
-        return true;
-      }
-      if (url.replace(/^http/, "https").startsWith(this.origin())) {
-        return true;
-      }
-      if (url.replace(/^https/, "http").startsWith(this.origin())) {
-        return true;
-      }
+    if (!url?.length) {
+      return false;
     }
-    return false;
+
+    if (url.startsWith("//")) {
+      url = "http:" + url;
+    }
+
+    if (!url.startsWith("http://") && !url.startsWith("https://")) {
+      return true;
+    }
```

Comment:
What about people linking to things like `file://` or `ftp://` URLs. I know it's weird, but we do get reports of using these kind of links on Meta.

Reply:
ah, there is a regression after all… 🥲 I'll restore my protocol tests and fix it